server/src/main/java/com/decathlon/ara/cartography/SquashExporter.java

```diff
@@ -27,8 +54,117 @@ public String getFormat() {
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_CRITICALITY");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_CATEGORY");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_STATUS");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_DESCRIPTION");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_#_TC");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_#_ATTACHEMENT");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_CREATED_ON");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_CREATED_BY");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_LAST_MODIFIED_ON");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_LAST_MODIFIED_BY");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_MILESTONE");
+        this.addCellToRow(header, rowIdx, "REQ_VERSION_CUF_<code du cuf>");
+        return header;
+    }
+
+    private void addRowToSheet(Sheet sheet, int index, String projectName, String path, String comment, String criticity, Date creationDate, String username) {
+        Row row = sheet.createRow(index);
+        this.addCellToRow(row, 0, "C");
+        this.addCellToRow(row, 2, projectName);
+        this.addCellToRow(row, 3, path);
+        this.addCellToRow(row, 4, "1");
+        this.addCellToRow(row, 5, "1");
+        this.addCellToRow(row, 7, criticity);
+        this.addCellToRow(row, 8, "CAT_UNDEFINED");
+        this.addCellToRow(row, 9, "WORK_IN_PROGRESS");
+        this.addCellToRow(row, 10, comment);
+        this.addCellToRow(row, 13, creationDate);
+        this.addCellToRow(row, 14, username);
+        this.addCellToRow(row, 15, creationDate);
+        this.addCellToRow(row, 16, username);
+    }
+
+    private void addCellToRow(Row row, int position, String value) {
+        Cell result = row.createCell(position);
+        result.setCellValue(value);
+    }
+
+    private void addCellToRow(Row row, int position, Date value) {
+        Cell result = row.createCell(position);
+        Workbook workbook = row.getSheet().getWorkbook();
+        CreationHelper creationHelper = workbook.getCreationHelper();
+        CellStyle dateStyle = workbook.createCellStyle();
+        dateStyle.setDataFormat(creationHelper.createDataFormat().getFormat("d/m/yy"));
+        result.setCellStyle(dateStyle);
+        result.setCellValue(value);
+    }
+
+    private String getFunctionalityPath(FunctionalityDTO functionality, List<FunctionalityDTO> functionalities) {
+        if (null == functionality.getParentId()) {
+            return "";
+        }
+        Long parentId = functionality.getParentId();
+        FunctionalityDTO parent = functionalities.stream()
+                .filter(f -> f.getId().equals(parentId))
+                .findFirst()
+                .orElse(null);
+        if (parent == null) {
+            return "";
+        }
+        return this.getFunctionalityPath(parent, functionalities) + "/" + parent.getName();
```

Comment:
String.format ?

Reply:
We're not in a loop here, so Java is able to detect the concatenation here and optimize it to a StringBuilder in the Bytecode.

And I found the String.format syntax less lisible than this (and again, String.format is absent of Java)